core/src/main/scala/quasar/plugin/postgres/CsvCreateSink.scala

```diff
@@ -77,18 +78,34 @@ object CsvCreateSink extends Logging {
       totalBytes <- Ref[F].of(0L)
       startAt <- timer.clock.monotonic(MILLISECONDS)
 
-      doCopy =
+      tempTbl = "precog_temp_" + tbl
+
+      doCopyToTemp =
         data
           .chunks
           .evalTap(recordChunks[F](totalBytes, log))
           // TODO: Is there a better way?
           .translate(Effect.toIOK[F] andThen LiftIO.liftK[CopyManagerIO])
-          .through(copyToTable(tbl, columns))
+          .through(copyToTable(tempTbl, columns))
 
       colSpecs <- columns.traverse(columnSpec).fold(
         invalid => AE.raiseError(new ColumnTypesNotSupported(invalid)),
         _.pure[F])
 
+      verifyExistence =
+        writeMode match {
+          case WriteMode.Create =>
+            checkExists(log)(tbl, schema) flatMap { result =>
+              if (result.exists(_ == 1))
```

Comment:
Could this value ever be `> 1`?

Reply:
It seems technically possible when `schema == None` if two tables in different schemas have the same name.

Reply:
Right, I'll switch this to `>=1`. Though, is that correct? Actually, I think what we may want here is to check that the `table_schema` is `'public'` when `schema == None`. Otherwise we could incorrectly return `true` here, if there is a table of the relevant name in a _different_ schema but not in `public`.

Reply:
Ok, pushed up a commit to address this - there were also a couple little bugs.